code/clue.js

```diff
@@ -29,36 +182,120 @@ const suspects = [
+  lounge,
+  ballroom,
+  hall,
+  spa,
+  livingRoom,
+  observatory,
+  theater,
+  guestHouse,
+  patio
+]
+console.log(rooms)
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FOR THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
 
-// FINISH THIS FUNCTION TO SHOW ALL INFORMATION ABOUT THE KILLER.
-// This function will be invoked when you click on the killer card.
+const mystery = {
+
+}
+
+
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer.
-  const theKiller = document.getElementById("killer")
-  const theKillerName = document.getElementById("killerName")
+  mystery.killer = randomSelector(suspects);
+
+  const theKiller = document.getElementById("killer");
+  const theKillerColor = document.getElementById("killerColor");
+  const theKillerName = document.getElementById("killerName");
+  const theKillerAge = document.getElementById("killerAge");
+  const theKillerOccupation = document.getElementById("killerOccupation");
+  const theKillerImage = document.getElementById("killerImage");
+  const theKillerDescription = document.getElementById("killerDiscription");
+
+  theKiller.style.background = "#a98";
+
+  theKillerName.innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+
+  theKillerAge.innerHTML = mystery.killer.age;
+
+  theKillerOccupation.innerHTML = mystery.killer.occupation;
+
+  theKillerImage.src = mystery.killer.image;
+
+  theKillerDescription.innerHTML = mystery.killer.description;
+
+  theKillerDescription.style.background = mystery.killer.color;
+}
+
+const pickWeapon = () => {
+
+  mystery.weapon = randomSelector(weapons);
+
+  const theWeapon = document.getElementById("weapon");
+  const theWeaponName = document.getElementById("weaponName");
+  const theWeaponMaterial = document.getElementById("weaponMaterial");
+  const theWeaponWeight = document.getElementById("weaponWeight");
+  const theWeaponImage = document.getElementById("weaponImage");
+
+  theWeapon.style.background = "#a98";
+
+  theWeaponName.innerHTML = mystery.weapon.name;
+
+  theWeaponMaterial.innerHTML = mystery.weapon.material;
+
+  theWeaponWeight.innerHTML = `${mystery.weapon.weight} kg`;
+
+  theWeaponImage.src = mystery.weapon.image;
+}
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms);
+
+  const theRoom = document.getElementById("rooms");
+  const theRoomName = document.getElementById("roomName");
+  const theRoomImage = document.getElementById("roomImage");
+
+  theRoom.style.background = "#a98";
+
+  theRoomName.innerHTML = `${mystery.room.name}`;
```

Comment:
Hm, I don't think that you need to it this way, just typing `mystery.room.name` should be enough